src/pages/index.js

```diff
@@ -35,36 +39,92 @@ addCardPopup.setEventListeners();
 
 const cardSection = new Section(
   {
-    items: initialCards,
     renderer: (item) => {
       const card = createCard(item);
       cardSection.addItem(card);
     },
   },
   ".cards__list"
 );
-cardSection.renderItems();
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeIcon
+  );
   const cardElement = card.getView();
   return cardElement;
 }
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
-  jobSelector: ".profile__description",
+  aboutSelector: ".profile__description",
+  avatarSelector: "#profile-image",
 });
 
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "965628aa-8b9c-448e-a4df-9a27d86f83f8",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getUserInfo()
+  .then((result) => {
+    userInfo.setUserInfo(result);
+    userInfo.setAvatar(result);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+api
+  .getInitialCards()
+  .then((result) => {
+    cardSection.renderItems(result);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
 function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo(formData);
-  profileEditPopup.close();
+  profileEditPopup.setSubmitButtonText("Saving...");
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```